### ISSUE TITLE
Google chart API to display location

### ISSUE THREAD

**Issue by v2belleville** [artifex404/wp-power-stats#4]
Hey,

I was very happy to find this pluggin : it’s simple, does what I need and says “No third party services”, which I think is important to preserve my visitors privacy.

Then I looked into the code to figure out a way to display the map on the web site.

Here is what I found
[google.vizualisation](http://v2belleville.eu/MesImages/ctrb/wp-power-stats-google.visualization.PNG)
Looks like a google chart API doesn’t it ?

And here is what google says about it’s API :
[Google Chart API](http://v2belleville.eu/MesImages/ctrb/google%20chart%20API.PNG)
[https://developers.google.com/chart/interactive/faq](https://developers.google.com/chart/interactive/faq)

In other words, the so called “No third party services” plugin sends the number of visit and the number of page view of your site to google servers …

Why not use a self hosted solution or an open Source solution ?

Regards

**Comment by artifex404** [comment]
Hello v2belleville,

Yes, you are correct. Currently the plugin does use Google Vizualization package to display the map.

It does use **Google VIsualization API** which does not need data being sent to the Google Server.

**Google Chart API** does need an URL being sent to Google for visualization and that is not being used.

The goal is to keep the plugin independent from any 3rd party services and ideally ship all needed vendor libraries so no CDN need to be used. That's still not completely implemented.

Currently the development of the plugin is on hold due to too many other projects and work. All pull requests and contributions are greatly appreciated to move this project forward.

**Comment by v2belleville** [comment]
Hi Igor,

 

Thanks for the quick reply.

 

Would you be able to send me a link to some documentation explaining how Google VIsualization works ?

 

Does the plugin currently use any other 3rd party service or CDN if yes, can you specify which one(s) ?

 

Regards

 

Véronique

 

 

De : Igor Buyanov [mailto:notifications@github.com] 
Envoyé : mardi 4 avril 2017 09:00
À : artifex404/wp-power-stats
Cc : v2belleville; Author
Objet : Re: [artifex404/wp-power-stats] Google chart API to display location (#4)

 

Hello v2belleville,

Yes, you are correct. Currently the plugin does use Google Vizualization package to display the map.

It does use Google VIsualization API which does not need data being sent to the Google Server.

Google Chart API does need an URL being sent to Google for visualization and that is not being used.

The goal is to keep the plugin independent from any 3rd party services and ideally ship all needed vendor libraries so no CDN need to be used. That's still not completely implemented.

Currently the development of the plugin is on hold due to too many other projects and work. All pull requests and contributions are greatly appreciated to move this project forward.

**Comment by artifex404** [comment]
Hello Véronique,

WP Power Stats does use any 3rd party services. It only loads 2 libraries from a CDN:

* jQuery library
* Google Charts library

More information about Google Charts library can be found here: https://developers.google.com/chart/

Note, that the plugin does **not** use Google Chart API, it does use Google Visualization API. https://developers.google.com/chart/interactive/docs/reference

**Comment by v2belleville** [comment]
Hi Igor,

 

Thanks for the input, I will look into it. The person I’ve installed power stat for is Google allergic so I have to know exactly how it’s used and make sure nothing is send to their servers.

 

If so I do hope to have time to add a few features and in that case will definitely send a pull request.

> Note, that the plugin does not use Google Chart API, it does use Google Visualization API.  <https://developers.google.com/chart/interactive/docs/reference> https://developers.google.com/chart/interactive/docs/reference

So you need to use de Google Charts library CDN in order to use google.visualisation , even though you do not use Google Chart API ? Is that correct ?

 

Cheers

 

V

 

De : Igor Buyanov [mailto:notifications@github.com] 
Envoyé : mercredi 12 avril 2017 14:08
À : artifex404/wp-power-stats
Cc : v2belleville; Author
Objet : Re: [artifex404/wp-power-stats] Google chart API to display location (#4)

 

Hello Véronique,

WP Power Stats does use any 3rd party services. It only loads 2 libraries from a CDN:

*	jQuery library
*	Google Charts library

More information about Google Charts library can be found here: https://developers.google.com/chart/

Note, that the plugin does not use Google Chart API, it does use Google Visualization API. https://developers.google.com/chart/interactive/docs/reference
